### PR TITLE
CI

### PR DESCRIPTION
**Pull request by grahamgilbert**
This is the start of using CI to perform the tests that the lovely @sheagcraig started. I am also starting to lint to make myself a neater coder. I ran the project through autopep8, which will hopefully not break anything.

### CODE REVIEW

**Comment by erikng** [comment]
Pep8 makes me really happy.

**Comment by grahamgilbert** [comment]
This is a first pass. It shows that `checkin` needs to be refactored so we can get the complexity number down to a sane level. I `noqa`-ed a lot of places where I probably shouldn't have done.

File: search/management/commands/search_maintenance.py [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
 from django.conf import settings
-from datetime import datetime, timedelta, date
+from datetime import datetime, timedelta, date  # noqa: F811
 import django.utils.timezone
 from server.models import *
 from search.models import *
 from inventory.models import *
 import server.utils as utils
-import datetime
+import datetime  # noqa: F811
```
Comment (epackorigan) [inline search/management/commands/search_maintenance.py]: overlap between this one and line 7 above. 
This file appears to only use 'datetime' and 'timedelta' also (so maybe remove 'date' from line 7?)

**Comment by sheagcraig** [comment]
All done going through your commits @grahamgilbert.

I had a few notes of things I wanted to clean up and PR to this too. Hopefully there's some useful suggestions in my comments above.

There was only the two things that I saw that stood out as a potential mistake or anything that would cause problems or difference in behavior.
1. Omission of `os_family` in search/views.py code.
2. No `view` identifier would cause an exception if it ever executed.

**Comment by sheagcraig** [comment]
I've got one more for you.

**Pull request merged**